### ISSUE TITLE
EC2 instances sometime show up with `healthState: Unknown` when SQL is used for clouddriver

### ISSUE THREAD

**Issue by costimuraru** [spinnaker/spinnaker#5061]
### Issue Summary:
We're using clouddriver with SQL (AuroraDB) for caching. We're noticing that some of our EC2 server groups show up as blue (with healthState: Unknown). When switching to using Redis, they show up fine (with color Green and healthState: Up).

### Cloud Provider(s):
AWS

### Feature Area:
clouddriver

### Description:
We're creating an app and deploy it to AWS EC2.
The app is deployed behind a load balancer (ELB).
When deploying the app, the instance remain in the Blue state. Investigating further, it looks like the UI is making a call to `https://gate-endpoint/applications/testaws2/serverGroups`

This contains the following snippet:
```
"instances": [
  {
    "availabilityZone": "us-east-1b",
    "health": [
      {
      "state": "Unknown",
      "type": "Amazon"
      }
    ],
    "healthState": "Unknown",
    "id": "i-027a4fbd868fb70f5",
    "launchTime": 1570732736000,
    "name": "i-027a4fbd868fb70f5"
  }
]
```

The healthState is **Unknown**, even though the instance is InService inside the load balancer (checked via AWS Console).

When disabling SQL, and switching to using Redis for clouddriver, the issue is no longer present.
The response for the aforementioned call also changes:

```
"instances": [
  {
    "availabilityZone": "us-east-1b",
    "health": [
      {
      "state": "Unknown",
      "type": "Amazon"
      },
      {
        "loadBalancers": [
          {
          "description": "N/A",
          "healthState": "Up",
          "name": "testaws-stage-elb",
          "state": "InService"
          }
        ],
        "state": "Up",
        "type": "LoadBalancer"
      }
    ],
    "healthState": "Up",
    "id": "i-027a4fbd868fb70f4",
    "launchTime": 1570732736000,
    "name": "i-027a4fbd868fb70f4"
  }
]
```

One theory we're having is that this happens when using SQL for clouddriver and the app name mismatches the load balancer name (eg. `testaws2` vs `testaws-stage-elb`).

### Steps to Reproduce:
1) Enable SQL for clouddriver
2) Create app1
3) Create app2
4) Create loadbalancer for app2 (eg app2-stage-elb)
5) Create EC2 deployment pipeline on app1 and hook it to app2-stage-elb

**Comment by asher** [comment]
@costimuraru I'm unable to reproduce:

![cross-app-elb](https://user-images.githubusercontent.com/59994/68047317-52c0ac80-fc9b-11e9-82e8-4c42b54a7589.jpg)

In this case, the serverGroup in app1 properly shows health from the ELB created under app2. When viewing the ELB from app2's Load Balancers tab, it also correctly shows health for the app1 serverGroup.

**Comment by asher** [comment]
Followed up on slack and requested some additional information. Once repro'd again, given a specific ec2 instance that isn't reflecting health from a cross-app ELB, I'd like to see the output of:
```
select * from cats_v1_instances_rel where id like "%$instanceId";
select * from cats_v1_loadBalancers_rel where id like "%loadbalancername%";
```

**Comment by asher** [comment]
Found the problem. I wasn't able to repro because netflix runs with edda enabled, which alters how CLB instance health is cached. A normal OSS install of spinnaker will cache this data via `AmazonLoadBalancerInstanceStateCachingAgent` but when edda is enabled, `EddaLoadBalancerCachingAgent` is used. Should have a fix next week.

**Comment by costimuraru** [comment]
So happy to hear that! Many thanks for taking the time to track and fix
this! Life saver indeed

**Comment by asher** [comment]
Should be fixed in master now and will land in the 1.17.1 release.
